l10n_th_tin_data/models/res_partner.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2020 Poonlap V. <poonlap@tanabutr.co.th>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+import logging
+import pprint
+import re
+from collections import OrderedDict
+from logging import INFO
+from os import path
+from pathlib import Path
+
+import requests
+from requests import Session
+from zeep import Client, Transport, helpers
+
+from odoo import _, api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    branch = fields.Char(
+        string="Tax Branch", help="Branch ID, e.g., 0000, 0001, ...", default="00000"
+    )
+
+    @staticmethod
+    def check_rd_tin_service(tin):
+        """Return bool after verifiying Tax Identification Number (TIN)
+           or Personal Identification Number (PIN)
+           by using Revenue Department's web service to prevent forging.
+           :param tin: a string for TIN or PIN
+        """
+
+        tin_web_service_url = (
+            "https://rdws.rd.go.th/serviceRD3/checktinpinservice.asmx?wsdl"
+        )
+        sess = Session()
+        mod_dir = path.dirname(path.realpath(__file__))
+        cert_path = str(Path(mod_dir).parents[0]) + "/static/cert/adhq1_ADHQ5.cer"
```

Comment:
May I ask. How do you get this cert? adhq1_ADHQ5.cer
Can we use it publicly?

Reply:
I found that RD has published this cert on their TIN service page [here](https://www.rd.go.th/publish/42533.0.html).
However, the CA root is not trusted and it will only be valid to 5/5/2565.

Reply:
However, the filename has been modified. I think for tracking and security purposes, the filename should remain the same as RD published.